Source/AssetRipper.Export.Modules.Shader/IO/SerializedExtensions.cs

```diff
@@ -576,7 +616,7 @@ public static void Export(this ShaderSubProgramBlob _this, ShaderWriter writer,
 				}
 
 				// we don't know shader type so pass vertex
-				_this.SubPrograms[subIndex].Export(writer, ShaderType.Vertex);
+				_this.GetSubProgram(writer.Shader.Collection, (uint)subIndex).Export(writer, ShaderType.Vertex);
```

Comment:
I think we should just store the collection on the blob instead of passing it as a parameter.